### DISCUSSION TITLE
Laravel Passport two guards

### DISCUSSION THREAD

**Discussion by Floppy012** [laravel/framework#50400] [Q&A]
I have two guards in use for two different user models. Passport supports multiple user types but only allows to use one guard for the auth controller. Is there any way to define different guards per client or maybe somehow combine the two guards?

Basically I want to be able to call `/oauth/authorize` when logged in as either one of my two user types. Currently only one or the other (depending on the guard set in the `passport.guard` config) is possible.

**Comment by ModulesSoft** [comment]
You may need to use roles and permissions approach by implementing gates or policies instead. 
Guards are used in rare cases and you should avoid using them if possible.

Learn more about Gates/Policies here:
https://laravel.com/docs/authorization#introduction
